docs/proposals/dynamic-openshift-settings-configmaps.md

```diff
@@ -0,0 +1,70 @@
+# Changing OpenShift settings dynamically through ConfigMaps
+
+## Problem
+
+Currently, there are settings contained withing the master-config.yaml file that
+are of value to an unpriviledged user. Specifically, the `autoProvisionEnabled`
+boolean under `jenkinsPipelineConfig` is only available to users with the
+ability to write to the master-config.
+
+Handling settings like this through project annotations was explored, but
+because unpriviledged users cannot annotate their own projects, it was
+abandoned.
```

Comment:
move this paragraph to an "other solutions considered" section, and add another item to describe the "using a single configmap with a specific name" option, the downsides of which are:

1) prevents users from ever using that name for anything else
2) doesn't allow for grouping of different settings into different configmaps

and the upsides of which are:
1) makes the code to process the project settings simpler (do not need to retrieve all configmaps and filter down to the ones w/ the annotation, and merge keys across configmaps, deal w/ conflicts, etc)